lang/typescript/src/utils/address-fields.ts

```diff
@@ -46,7 +47,13 @@ export function splitAddressField(
   fieldDefinition: FieldConcatenationRule[],
   concatenatedAddress: string,
 ): Partial<Address> {
-  const values = concatenatedAddress.split(RESERVED_DELIMITER);
+  // Support custom delimiteres defined on ANY sub-field definition
+  const customDelimiter = fieldDefinition.find(
+    (def) => def.delimiter !== undefined,
+  )?.delimiter;
+  const values = concatenatedAddress.split(
+    customDelimiter ?? RESERVED_DELIMITER,
+  );
```

Comment:
Here handles the custom delimiter. Based on my current implementation I can't really support multiple delimiters so I'm just getting the first custom one I can find in the language override.